### PR TITLE
Add neg_op kernel and unittest

### PR DESCRIPTION
**Pull request by levi131**
### PR types
New features
### PR changes
OPs

### Describe
Add neg op and test.

### CODE REVIEW

File: paddle/fluid/operators/neg_op.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+class NegOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "(Tensor), The input tensor of neg op.");
+    AddOutput("Out", "(Tensor), The output tensor of neg op.");
+    AddComment(R"DOC(
+Neg Operator.
+
+This operator is used to perform elementwise neg for input $X$.
+$$out = |x|$$
```
Comment (Aurelius84) [inline paddle/fluid/operators/neg_op.cc]: Here  should be `out  = -x`

File: paddle/fluid/operators/neg_op.cc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    retv->SetOutput(framework::GradVarName("X"), this->InputGrad("X"));
+  }
+};
+
+class NegOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    OP_INOUT_CHECK(ctx->HasInput("X"), "Input", "X", "neg");
```
Comment (Aurelius84) [inline paddle/fluid/operators/neg_op.cc]: "neg" -> "Neg"

File: paddle/fluid/operators/neg_op.cc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+class NegGradOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    OP_INOUT_CHECK(ctx->HasInput(framework::GradVarName("Out")), "Input",
+                   "Out@Grad", "NegGrad");
+    OP_INOUT_CHECK(ctx->HasOutput(framework::GradVarName("X")), "Output",
+                   "X@Grad", "AbsGrad");
```
Comment (Aurelius84) [inline paddle/fluid/operators/neg_op.cc]: "AbsGrad" -> "NegGrad"

File: paddle/fluid/operators/neg_op.cu [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/operators/neg_op.h"
+#include "paddle/fluid/platform/float16.h"
+
+namespace ops = paddle::operators;
```
Comment (Aurelius84) [inline paddle/fluid/operators/neg_op.cu:54]: Could we consider a CUDAKernel implementation to speed up the calculation in the future?

File: paddle/fluid/operators/neg_op.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        ctx.Output<framework::Tensor>(framework::GradVarName("X"));
+
+    auto numel = d_out->numel();
+    auto* dout_data = d_out->data<math::Real<T>>();
+    auto* dx_data = d_x->mutable_data<T>(
+        ctx.GetPlace(), static_cast<size_t>(numel * sizeof(T)));
+
+    auto& dev_ctx = ctx.template device_context<DeviceContext>();
+    platform::ForRange<DeviceContext> for_range(dev_ctx, numel);
+    math::NegFunctor<T> functor(dout_data, dx_data, numel);
```
Comment (Aurelius84) [inline paddle/fluid/operators/neg_op.h:61]: Please make sure the calculation is right. I thought `dy/dx = -1`

File: python/paddle/fluid/tests/unittests/test_neg_op.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import paddle
+from op_test import OpTest
+
+paddle.enable_static()
+
+
+class TestNegOp(OpTest):
+    def setUp(self):
+        self.op_type = 'neg'
+        self.dtype = np.float64
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/test_neg_op.py]: Could you and more unittest to test `float, int, int64, float16`

File: paddle/fluid/operators/neg_op.cc (hunk 1 above)
Comment (levi131) [reply paddle/fluid/operators/neg_op.cc]: modified

File: paddle/fluid/operators/neg_op.cu (hunk 4 above)
Comment (levi131) [reply paddle/fluid/operators/neg_op.cu:54]: I write a cuda kernel for forward process of neg op.

**Review by Aurelius84 — DISMISSED** [review]
LGTM

**Review by Aurelius84 — DISMISSED** [review]

**Review by Aurelius84 — APPROVED** [review]

**Comment by levi131** [comment]
This pr is closed, and I will submit another pr in which interface "paddle.neg" is implemented by interface "paddle.scale()".

**Pull request closed without merging**